stdlib/Dates/src/periods.jl

```diff
@@ -451,11 +451,11 @@ toms(c::Second)      = 1000 * value(c)
 toms(c::Minute)      = 60000 * value(c)
 toms(c::Hour)        = 3600000 * value(c)
 toms(c::Period)      = 86400000 * days(c)
-toms(c::CompoundPeriod) = isempty(c.periods) ? 0.0 : Float64(sum(toms, c.periods))
+toms(c::CompoundPeriod) = isempty(c.periods) ? 0.0 : Float64(sum(toms, c.periods))::Float64
```

Comment:
```suggestion
toms(c::CompoundPeriod) = isempty(c.periods) ? 0.0 : 
    sum(convert(toms(p),Float64)::Float64, c.periods)
```
would be slightly more efficient since it makes `sum` type stable also?

Reply:
Hi @aviatesk !

You mean something like this right: `sum(p -> convert(Float64, toms(p))::Float64, c.periods)`

In this case, I tested here and it is indeed faster. I will update the PR. Thanks!